lib/auth/auth.go

```diff
@@ -2122,6 +2123,34 @@ func (a *Server) GetNodes(ctx context.Context, namespace string, opts ...service
 	return a.GetCache().GetNodes(ctx, namespace, opts...)
 }
 
+// ListNodes returns a paginated list of nodes.
+func (a *Server) ListNodes(ctx context.Context, namespace string, limit int, startKey string) (page []types.Server, nextKey string, err error) {
+	nodes, err := a.GetCache().GetNodes(ctx, namespace)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// Find startKey
+	startIndex := sort.Search(len(nodes), func(i int) bool {
+		return nodes[i].GetName() >= startKey
+	})
```

Comment:
This works only on the condition that the nodes returned by `GetNodes` are ordered lexicographically by their UUIDs. Can this be assumed with all our backend implementations?

Reply:
Yes, it looks like our backend interface depends on it, all of the backends we support order the return of `GetRange` manually if needed.